autosklearn/data/validation.py

```diff
@@ -12,6 +14,21 @@
 from autosklearn.util.logging_ import get_named_client_logger
 
 
+def convert_if_sparse(y: SUPPORTED_TARGET_TYPES) -> SUPPORTED_TARGET_TYPES:
```

Comment:
Don't we need a new data type annotation for the target which isn't sparse any more?

Reply:
Yes we do, i suppose using 
* For types passed around internally `SUPPORTED_TARGET_TYPES = Union[List, Series, DataFrame, np.ndarray]`
* For external api points, the type is `Union[spmatrix, SUPPORTED_TARGET_TYPES]`